### ISSUE TITLE
setColumnOrder does not work if 2 header rows

### ISSUE THREAD

**Issue by jcgueriaud1** [vaadin/vaadin-grid-flow#827]
In Vaadin 14.1.0.beta2, if I call setColumnOrder on a grid with 2 header rows then I've got an exception: ConcurrentModificationException (see the stack trace at the bottom)

This error only appears in Java 9+ (tested with openjdk 11 and 12) but could be a problem in java 8 fixed in java 9+( https://stackoverflow.com/questions/54824656/since-java-9-hashmap-computeifabsent-throws-concurrentmodificationexception-on )

Here the code to reproduce the problem:

```

@Route("grid2")
public class Grid2View extends VerticalLayout {

    private Grid<Person> multiSelectGrid = new Grid<>(Person.class);

    public Grid2View() {
        HeaderRow headerRow = multiSelectGrid.appendHeaderRow();
        multiSelectGrid.setColumnReorderingAllowed(true);
        add(multiSelectGrid);

        Button button = new Button("set same order");
        button.addClickListener(event -> {
            multiSelectGrid.setColumnOrder(multiSelectGrid.getColumns());
        });

        add(button);
    }

    public static class Person {
        private String name;

        public String getFirstName() {
            return firstName;
        }

        public void setFirstName(String firstName) {
            this.firstName = firstName;
        }

        private String firstName;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }
}
```
---------------------
```
java.util.ConcurrentModificationException
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
	at com.vaadin.flow.component.grid.GridColumnOrderHelper$GraphNodeLeafCache.getColumnIDs(GridColumnOrderHelper.java:292)
	at com.vaadin.flow.component.grid.GridColumnOrderHelper$GraphNodeLeafCache.findFirstContaining(GridColumnOrderHelper.java:312)
	at com.vaadin.flow.component.grid.GridColumnOrderHelper.reorderColumnsAndConsumeIDs(GridColumnOrderHelper.java:187)
	at com.vaadin.flow.component.grid.GridColumnOrderHelper.setColumnOrder(GridColumnOrderHelper.java:85)
	at com.vaadin.flow.component.grid.Grid.setColumnOrder(Grid.java:3825)
	at org.vaadin.jeanchristophe.Grid2View.lambda$new$9b1b5227$1(Grid2View.java:33)
```

**Comment by petrixh** [comment]
PR: #828

**Comment by tomivirkki** [comment]
Fixed by #828
